static/app/components/events/viewHierarchy/wireframe.tsx

```diff
@@ -0,0 +1,146 @@
+import {useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+import {useResizeObserver} from '@react-aria/utils';
+
+import {ViewHierarchyWindow} from 'sentry/components/events/viewHierarchy';
+import {useMousePan} from 'sentry/components/events/viewHierarchy/useMousePan';
+
+type Rect = Pick<ViewHierarchyWindow, 'x' | 'y' | 'width' | 'height'>;
+
+export function getCoordinates(hierarchies: ViewHierarchyWindow[]) {
+  return hierarchies.map(hierarchy => {
+    const newHierarchy: Rect[] = [];
+
+    const nodesToProcess = [hierarchy];
+    while (nodesToProcess.length) {
+      const node = nodesToProcess.pop();
+
+      if (!node) {
+        continue;
+      }
```

Comment:
I'm not sure this line is necessary, is there ever a case where node is not defined? Seems like node is of `ViewHierarchyWindow` type and it's also impossible to enter here if `nodesToProcess.length` < 1

Reply:
Typescript was complaining here because I think the return type of `pop()` is `T | undefined`. That `undefined` should've been ignored by the `while` condition but I guess it doesn't properly resolve. I went between this and `!` at the end of `pop()`. I think I'll use `!` pop since you're right, it shouldn't be possible to get undefined for `node` and it's a little distracting to have this if statement